packages/material-ui-lab/src/SliderStyled/SliderStyled.js

```diff
@@ -0,0 +1,353 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useThemeProps, muiStyled, fade, lighten, darken } from '@material-ui/core/styles';
+import { propsToClassKey } from '@material-ui/styles';
+import { capitalize } from '@material-ui/core/utils';
+import SliderUnstyled from '../SliderUnstyled';
+
+const overridesResolver = (props, styles, name) => {
+  const {
+    color = 'primary',
+    marks: marksProp = false,
+    max = 100,
+    min = 0,
+    orientation = 'horizontal',
+    step = 1,
+    track = 'normal',
+  } = props;
+
+  const marks =
+    marksProp === true && step !== null
+      ? [...Array(Math.floor((max - min) / step) + 1)].map((_, index) => ({
+          value: min + step * index,
+        }))
+      : marksProp || [];
+
+  const marked = marks.length > 0 && marks.some((mark) => mark.label);
+
+  const styleOverrides = {
+    ...styles.root,
+    ...styles[`color${capitalize(color)}`],
+    '&.Mui-disabled': styles.disabled,
+    ...(marked && styles.marked),
+    ...(orientation === 'vertical' && styles.vertical),
+    ...(track === 'inverted' && styles.trackInverted),
+    ...(track === false && styles.trackFalse),
+    [`.& ${name}-rail`]: styles.rail,
+    [`.& ${name}-track`]: styles.track,
+    [`.& ${name}-mark`]: styles.mark,
+    [`.& ${name}-markLabel`]: styles.markLabel,
+    [`.& ${name}-valueLabel`]: styles.valueLabel,
+    [`.& ${name}-thumb`]: {
+      ...styles.thumb,
+      ...styles[`thumbColor${capitalize(color)}`],
+      '&.Mui-disabled': styles.disabled,
+    },
+  };
+
+  return styleOverrides;
+};
+
+const variantsResolver = (props, styles, theme, name) => {
+  const { state = {} } = props;
+  let variantsStyles = {};
+  if (theme && theme.components && theme.components[name] && theme.components[name].variants) {
+    const themeVariants = theme.components[name].variants;
```

Comment:
```suggestion
  const themeVariants = theme?.components?.[name]?.variants;
  if (themeVariants !== undefined) {
```

Optional chaining wants to get involved.

Reply:
Refactored 👍  Also noticed that this is a generic method, so I moved it to `muiStyled` instead of needing it to be defined in each component